packages/zoe/src/contracts/newSwap/getCurrentPrice.js

```diff
@@ -52,16 +52,22 @@ export const makeGetCurrentPrice = (
       // remainder to the pool to get a quote. Then we'll add the fee to deltaX
       // before sending out the quote.
 
-      // prelimProtocolFee will be replaced by protocolFee once we get the final
-      // value of amountIn from getPriceGivenAvailableInput.
-      const prelimProtocolFee = multiplyBy(amountIn, protocolFeeRatio);
-      const poolAmountIn = amountMath.subtract(amountIn, prelimProtocolFee);
+      // amountIn will be divided into deltaX (what's added to the pool) and the
+      // protocol fee. protocolFee will be protocolFeeRatio * deltaX.
+      // Therefore, amountIn = (1 + protocolFeeRatio) * deltaX, and
+      // protocolFee =  protocolFeeRatio * amountIn / (1 + protocolFeeRatio).
+      const feeOverOnePlusFee = makeRatio(
+        protocolFeeBP,
+        centralBrand,
+        add(BASIS_POINTS, protocolFeeBP),
+      );
+      const protocolFee = multiplyBy(amountIn, feeOverOnePlusFee);
+      const poolAmountIn = amountMath.subtract(amountIn, protocolFee);
       const price = getPool(brandOut).getPriceGivenAvailableInput(
         poolAmountIn,
         brandOut,
         poolFeeBP,
       );
-      const protocolFee = multiplyBy(price.amountIn, protocolFeeRatio);
 
       // price.amountIn is what the user pays, and includes the protocolFee. The
       // user will receive price.amountOut.
```

Comment:
> I think we agree that this is off by a tiny fraction, but won't cause errors.

Rounding fees down actually will cause errors, though. For tiny trades, the fee will be so low that there's a perverse incentive to do many tiny trades instead of the normal, larger one.

Reply:
Sorry, in a related conversation you made the distinction that this kind of thing wouldn't throw Errors.

I previously made the argument about incentive compatibility, so I'm sympathetic, but that's at a different scale than causing Errors.  I think  this fix can wait until we have an agreement on how to handle rounding ratio calculations. That will long before there is actual value at stake on our platform.